File: integration-tests-java/src/test/java/io/micronaut/jimmer/java/it/resource/TestErrorCodeTestCase.java
package io.micronaut.jimmer.java.it.resource;

import io.micronaut.test.extensions.junit5.annotation.MicronautTest;
import io.restassured.response.Response;
import io.restassured.specification.RequestSpecification;
import jakarta.inject.Inject;
import org.apache.http.HttpStatus;
import org.junit.jupiter.api.Assertions;
import org.junit.jupiter.api.Test;

@MicronautTest
public class TestErrorCodeTestCase {

    @Inject RequestSpecification requestSpecification;

    @Test
    void testHelloEndpoint() {
        Response response = requestSpecification.when().get("/bookResource/testError");
        Assertions.assertEquals(HttpStatus.SC_INTERNAL_SERVER_ERROR, response.statusCode());
        Assertions.assertEquals("USER_INFO", response.body().jsonPath().getString("family"));
        Assertions.assertEquals("ILLEGAL_USER_NAME", response.body().jsonPath().getString("code"));
        Assertions.assertEquals("[a]", response.body().jsonPath().getString("illegalChars"));
    }
}

File: integration-tests-kotlin/src/test/kotlin/io/micronaut/jimmer/kotlin/it/resource/TestErrorCodeTestCase.kt
package io.micronaut.jimmer.kotlin.it.resource

import io.micronaut.test.extensions.junit5.annotation.MicronautTest
import io.restassured.specification.RequestSpecification
import jakarta.inject.Inject
import org.apache.http.HttpStatus
import org.junit.jupiter.api.Assertions
import org.junit.jupiter.api.Test

@MicronautTest
class TestErrorCodeTestCase {
    @Inject
    lateinit var requestSpecification: RequestSpecification

    @Test
    fun testHelloEndpoint() {
        val response = requestSpecification.`when`().get("/bookResource/testError")
        Assertions.assertEquals(HttpStatus.SC_INTERNAL_SERVER_ERROR, response.statusCode())
        Assertions.assertEquals("USER_INFO", response.body().jsonPath().getString("family"))
        Assertions.assertEquals("ILLEGAL_USER_NAME", response.body().jsonPath().getString("code"))
        Assertions.assertEquals("[]", response.body().jsonPath().getString("illegalChars"))
    }
}
